hazelcast/include/hazelcast/client/serialization/generic_record_builder.h

```diff
@@ -0,0 +1,980 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <unordered_set>
+
+#include <boost/optional.hpp>
+#include <boost/any.hpp>
+#include <boost/variant.hpp>
+
+#include "hazelcast/client/serialization/pimpl/compact/schema_writer.h"
+#include "hazelcast/client/serialization/generic_record.h"
+#include "hazelcast/client/big_decimal.h"
+#include "hazelcast/client/local_time.h"
+#include "hazelcast/client/local_date.h"
+#include "hazelcast/client/local_date_time.h"
+#include "hazelcast/client/offset_date_time.h"
+#include "hazelcast/util/byte.h"
+
+namespace hazelcast {
+namespace client {
+namespace serialization {
+namespace generic_record {
+
+class HAZELCAST_API generic_record_builder
+{
+public:
+    /**
+     * Creates a Builder that will build a {@link generic_record} in
+     * Compact format with the given type name and schema:
+     * <pre>{@code
+     * generic_record record = generic_record_builder{"typeName"}
+     *                                 .set_string("name", "foo")
+     *                                 .set_int32("id", 123)
+     *                                 .build();
+     * }</pre>
+     *
+     * @param typeName of the schema
+     * @return generic_record_builder for Compact format
+     */
+    explicit generic_record_builder(std::string type_name);
+
+    /**
+     * @return a new constructed generic_record
+     * @throws exception::hazelcast_serialization when the generic_record cannot
+     * be built.
+     */
+    generic_record build();
+
+    /**
+     * It is legal to set the field again only when Builder is created with
+     * {@link generic_record::new_builder_with_clone()}. Otherwise, it is
+     * illegal to set to the same field twice.
+     *
+     * @param fieldName name of the field as it is defined in its schema
+     *                  definition.
+     * @param value     to set to generic_record
+     * @return itself for chaining
+     * @throws exception::hazelcast_serialization if the field name does not
+     * exist in the schema definition or the type of the field does not match
+     * the one in the schema definition or the same field is trying to be set
+     * without using
+     *                                            {@link
+     * generic_record::new_builder_with_clone()}.
+     */
+    generic_record_builder& set_boolean(std::string field, bool value);
```

Comment:
inconsistency between code and comment. field, fieldName

Reply:
Fixed.